src/hotspot/share/services/threadStackTracker.hpp

```diff
@@ -74,11 +75,12 @@ class ThreadStackTracker : AllStatic {
   static bool initialize(NMT_TrackingLevel level);
   static bool transition(NMT_TrackingLevel from, NMT_TrackingLevel to);
 
+  // Require ThreadCritical lock held
   static void new_thread_stack(void* base, size_t size, const NativeCallStack& stack);
   static void delete_thread_stack(void* base, size_t size);
 
   static bool   track_as_vm()  { return AIX_ONLY(false) NOT_AIX(true); }
```

Comment:
Unrelated to your patch: I wonder whether this could be controlled as a diagnostic switch, to be able to test both implementations on normal platforms.